parsers/ES.py

```diff
@@ -37,45 +21,107 @@
+    "eol": "production.wind",
+    "cc": "production.gas",
+    "vap": "production.oil",
+    "fot": "production.solar",
+    "sol": "production.solar",
+    "hid": "storage.hydro",
+    "car": "production.coal",
+    "resid": "production.biomass",
+    "termRenov": "production.unknown",
+    "cogenResto": "production.unknown",
+    "tnr": "production.unknown",
+    "trn": "production.unknown",
+    "otrRen": "production.unknown",
+    "cogen": "production.gas",
+    "cb": "exchange.pe_ma",
+    "icb": "exchange.pe_ma",
+    "emm": "exchange.ma_me",
+    "emi": "exchange.ma_ib",
+    "eif": "exchange.ib_fo",
+    "inter": "exchange.int",
+}
 
+API_CODE_MAPPING = {
+    ZoneKey("ES"): "DEMANDAQH",
+    ZoneKey("ES-CE"): "CEUTA5M",
+    ZoneKey("ES-CN-FVLZ"): "LZ_FV5M",
+    ZoneKey("ES-CN-GC"): "GCANARIA5M",
+    ZoneKey("ES-CN-HI"): "EL_HIERRO5M",
+    ZoneKey("ES-CN-IG"): "LA_GOMERA5M",
+    ZoneKey("ES-CN-LP"): "LA_PALMA5M",
+    ZoneKey("ES-CN-TE"): "TENERIFE5M",
+    ZoneKey("ES-IB-FO"): "FORMENTERA5M",
+    ZoneKey("ES-IB-IZ"): "IBIZA5M",
+    ZoneKey("ES-IB-MA"): "MALLORCA5M",
+    ZoneKey("ES-IB-ME"): "MENORCA5M",
+    ZoneKey("ES-ML"): "MELILLA5M",
+}
 
-ZONE_FUNCTION_MAP: Dict[ZoneKey, Callable] = {
-    ZoneKey("ES"): IberianPeninsula,
-    ZoneKey("ES-CE"): Ceuta,
-    ZoneKey("ES-CN-FVLZ"): LanzaroteFuerteventura,
-    ZoneKey("ES-CN-GC"): GranCanaria,
-    ZoneKey("ES-CN-HI"): ElHierro,
-    ZoneKey("ES-CN-IG"): Gomera,
-    ZoneKey("ES-CN-LP"): LaPalma,
-    ZoneKey("ES-CN-TE"): Tenerife,
-    ZoneKey("ES-IB-FO"): Formentera,
-    ZoneKey("ES-IB-IZ"): Ibiza,
-    ZoneKey("ES-IB-MA"): Mallorca,
-    ZoneKey("ES-IB-ME"): Menorca,
-    ZoneKey("ES-ML"): Melilla,
+ZONE_MAPPING = {
+    ZoneKey("ES"): "Peninsula",
+    ZoneKey("ES-CE"): "Peninsula",
+    ZoneKey("ES-CN-FVLZ"): "Canarias",
+    ZoneKey("ES-CN-GC"): "Canarias",
+    ZoneKey("ES-CN-HI"): "Canarias",
+    ZoneKey("ES-CN-IG"): "Canarias",
+    ZoneKey("ES-CN-LP"): "Canarias",
+    ZoneKey("ES-CN-TE"): "Canarias",
+    ZoneKey("ES-IB-FO"): "Baleares",
+    ZoneKey("ES-IB-IZ"): "Baleares",
+    ZoneKey("ES-IB-MA"): "Baleares",
+    ZoneKey("ES-IB-ME"): "Baleares",
+    ZoneKey("ES-ML"): "Peninsula",
 }
 
-EXCHANGE_FUNCTION_MAP: Dict[str, Callable] = {
-    "ES->ES-IB-MA": Mallorca,
-    "ES-IB-IZ->ES-IB-MA": Mallorca,
-    "ES-IB-FO->ES-IB-IZ": Formentera,
-    "ES-IB-MA->ES-IB-ME": Mallorca,
+TIMEZONES_MAPPING = {
+    ZoneKey("ES"): "Europe/Madrid",
+    ZoneKey("ES-CE"): "Africa/Ceuta",
+    ZoneKey("ES-CN-FVLZ"): "Atlantic/Canary",
+    ZoneKey("ES-CN-GC"): "Atlantic/Canary",
+    ZoneKey("ES-CN-HI"): "Atlantic/Canary",
+    ZoneKey("ES-CN-IG"): "Atlantic/Canary",
+    ZoneKey("ES-CN-LP"): "Atlantic/Canary",
+    ZoneKey("ES-CN-TE"): "Atlantic/Canary",
+    ZoneKey("ES-IB-FO"): "Europe/Madrid",
+    ZoneKey("ES-IB-IZ"): "Europe/Madrid",
+    ZoneKey("ES-IB-MA"): "Europe/Madrid",
+    ZoneKey("ES-IB-ME"): "Europe/Madrid",
+    ZoneKey("ES-ML"): "Africa/Ceuta",
 }
```

Comment:
These can probably be combined, either as a dict with the zoneKey as the key and appropriate subkeys for each of the values or as a named tuple.

Reply:
I think joining all in one tuple would make it more complicated and less self-explanatory, even though it would be less repetitive.